pilot/pkg/bootstrap/sidecarinjector.go

```diff
@@ -30,10 +31,15 @@ const (
 	webhookName = "sidecar-injector.istio.io"
 )
 
+var (
+	injectionEnabled = env.RegisterBoolVar("INJECT_ENABLED", true, "Enable mutating webhook handler.")
+)
+
 func (s *Server) initSidecarInjector(args *PilotArgs) (*inject.Webhook, error) {
+	// currently the constant: "./var/lib/istio/inject"
 	injectPath := args.InjectionOptions.InjectionDirectory
-	if injectPath == "" {
-		log.Infof("Skipping sidecar injector, injection path is missing")
+	if injectPath == "" || !injectionEnabled.Get() {
```

Comment:
why do we have two ways to disable this?

Reply:
We don't. InjectionDirectory is from a constant. I'm planning to remove it - and keep it in the code, there is no need to pass a constant all the way down from main as an option.

Reply:
Also making injection directory custom doesn't seem to add value - it's part of the mount or image, no need for users to set it to a different path.

Reply:
Oh I thought it was dynamic